src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,215 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { RequestOptions } from "http";
+import { IncomingMessage, ServiceClientCredentials, WebResource } from "ms-rest";
+import * as requestP from 'request-promise';
+import { URL } from "url";
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { DeployResult } from "vscode-azurekudu/lib/models";
+import { SiteTreeItem } from "./explorer/SiteTreeItem";
+import { ext } from './extensionVariables';
+import { delay } from "./utils/delay";
+import { openUrl } from './utils/openUrl';
+
+const detectorId: string = 'LinuxContainerStartFailure';
+enum ColumnName {
+    status = "Status",
+    message = "Message",
+    name = "Data.Name",
+    value = "Data.Value",
+    expanded = "Expanded",
+    solutions = "Solutions",
+    time = "Time",
+    instance = "Instance",
+    facility = "Facility",
+    failureCount = "FailureCount"
+}
+
+export async function checkLinuxWebAppDownDetector(node: SiteTreeItem): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('appService.linuxWebAppDownDetector', async (context: IActionContext): Promise<void> => {
+        const deployment: DeployResult = await node.root.client.kudu.deployment.getResult('latest');
+        if (!deployment.startTime) {
+            // if there's no deployment detected, nothing can be done
+            context.telemetry.properties.cancelStep = 'noDeployResult';
+            return;
+        }
+
+        const deployResultTime: Date = new Date(deployment.startTime);
+
+        const detectorOutput: string = `Diagnosing web app "${node.root.client.siteName}" for critical errors...`;
+        ext.outputChannel.appendLine(detectorOutput);
+
+        const detectorUri: string = `https://management.azure.com/subscriptions/${node.root.subscriptionId}/resourceGroups/${node.root.client.resourceGroup}/providers/Microsoft.Web/sites/${node.root.client.siteName}/detectors/${detectorId}`;
+        const requestOptions: WebResource & Partial<{ qs: queryString }> = new WebResource();
```

Comment:
This is the second or third time you've had to workaround the fact that the app service sdk doesn't support something. Can you create a separate utility function to help with these calls? Even if we don't expose it in the shared package, it can at least be a separate file that we can copy/paste between repos.

Also is there a bug to add this to the app service sdk?

For example, [here](https://github.com/microsoft/vscode-azuretools/blob/master/appservice/src/createAppService/SiteRuntimeStep.ts#L59) you correctly used `resourceManagerEndpointUrl`. In the above code, it will fail for other sovereigns.

Reply:
Still waiting on this. As an aside, maybe try `ServiceClient.sendRequest`

Reply:
This is resolved by Eric's changes in the azuretools https://github.com/microsoft/vscode-azuretools/commit/1d23d4f2c8126b9ff1ab7d70f8f1ca7c0266c9a9